# metrics.classification_report函数介绍

In [3]:
from sklearn.metrics import classification_report
import numpy as np

In [5]:
y_true = np.array([[0,0,1], [1,1,0],[0,1,0]])
y_pred = np.array([[0,1,1], [1,0,1],[1,0,0]])
res = classification_report(y_true, y_pred)
print(res)

              precision    recall  f1-score   support

           0       0.50      1.00      0.67         1
           1       0.00      0.00      0.00         2
           2       0.50      1.00      0.67         1

   micro avg       0.40      0.50      0.44         4
   macro avg       0.33      0.67      0.44         4
weighted avg       0.25      0.50      0.33         4
 samples avg       0.33      0.50      0.39         4



## 几个常见的模型指标介绍
以下都以label0来举例子。
1. precision: 预测1里有几个是真1   $$1/2 = 0.5$$
2. recall: 真1里有几个被预测为1  $$1/1 = 1.0$$
3. F1_score: 兼顾准确和召回，  
$$f1 = 2*\frac{precision*recall}{precision+recall}$$  
$$2*0.5*1/（0.5+1） = 0.67$$
4. micro avg: 微平均，表示所有类别中预测正确量与总样本量的比值  $$2/5 = 0.4$$
5. macro avg: 宏平均，表示所有类别对应指标的平均值  $$(0.5+0.0+0.5)/3 = 0.33$$
6. weighted avg: 带权重平均，表示类别样本占总样本的比重与对应指标的乘积的累加  $$(1*0.5+2*0+1*0.5)/4 = 0.25$$
7. samples avg: 样本平均  $$$$